docs/src/docs/asciidoc/end-to-end-maven-guide.adoc

```diff
@@ -0,0 +1,830 @@
+= End-to-end Native Maven Plugin guide
+The GraalVM team
+:highlighjsdir: {maven-relative-srcdir}/highlight
+
+
+This guide leads you through the process of integrating Native Maven Plugin with your project.
+It starts from adding the plugin to your project, goes through some of the main functionalities that you may use (like collecting the metadata),
+all the way to some diagnostics tools that you can use to analyse native executables you have made.
+You can see an example of tests on which we tested whole guide <<example,here>>
+
+In case you only want to see how a simple application works in practise, you can check <<quickstart-maven-plugin.adoc#,our demo>>.
```

Comment:
```suggestion
In case you only want to see how a simple application works in practise, you can check <<quickstart-maven-plugin.adoc#,our demo>>.
```